File: test/fixtures/php/Greeter.php
<?php

require_once 'Person.php';

class Greeter {
    private Person $person;

    public function __construct(Person $person) {
        $this->person = $person;
    }

    public function sayHello(): string {
        return "Greeter says: " . $this->person->greet();
    }
}

File: test/fixtures/rust/src/greeter.rs
use crate::Person;

pub struct Greeter {
    person: Person,
}

impl Greeter {
    pub fn new(person: Person) -> Self {
        Greeter { person }
    }

    pub fn say_hello(&self) -> String {
        format!("Greeter says: {}", self.person.greet())
    }
}
